src/Dialogs/Imap/ImapDialog.vala

```diff
@@ -338,7 +338,7 @@ public class OnlineAccounts.ImapDialog : Hdy.Window {
 
     public async void load_configuration (E.Source account_source, GLib.Cancellable? cancellable) throws Error {
         if (!account_source.has_extension (E.SOURCE_EXTENSION_MAIL_ACCOUNT)) {
-            throw new Camel.Error.ERROR_GENERIC (_("The data provided does not seem to reflect a valid mail account."));
+            throw new Camel.Error.ERROR_GENERIC (_("Something is wrong with the provided e-mail account data."));
```

Comment:
This wording seems less precise to me. The first sentence is saying that there doesn't seem to be an account that matches the data. The second sentence is much more vague about there being "something wrong", but it seems like the issue is more that maybe the email address or server URL is invalid

Reply:
@danrabbit How about "Correct the e-mail address first", or similar?
Validity isn't reflected, and "seem" is vague. "data provided" could be "provided data", but then why is it "data"?

Reply:
As I understand, this function throws an error when the account data returns a source that doesn't contain an email account. This could be that the account doesn't exist, but I suppose it could also mean that there is a collection account that exists on the server, but it's not an email account.

There's not any implied second step here, so I don't think "first" makes sense.

I'm not sure what "validity isn't reflected" means. Yes "seem" is vague and that's because there could probably be multiple causes for the issue. The only thing we know for sure is that the account source we got in this function doesn't contain a mail account and we're speculating about the cause